docs/dev_guide.md

```diff
@@ -1,31 +1,183 @@
 # Dev Guide
 
-## Adding a new param in GCSFuse
+## Contributing to GCSFuse
+
+**If you're interested in contributing a new feature, enhancement, or bug fix,
+please open a GitHub issue to discuss your proposal with the GCSFuse team first
+before making the contribution.** Make sure to describe the use case (what are
+you trying to solve and why?), and the proposed solution in the GitHub issue.
+
+**Note:** We make no guarantees that contribution requests are accepted, or will
+be rolled out as a Generally Available (GA) feature
+
+Rollout of any new feature should be done in 2 phases:
+
+**1. Experimental Phase:**
+
+* New features should initially be introduced behind an experimental flag. These
```

Comment:
Can we add make some text here to link to the section below of Adding a CLI/Config flag.